### PR TITLE
[Feature] Change completed tasks UI to checkbox

### PR DESCRIPTION
**Pull request by daiHash**
## Description
- Change completed tasks UI to checkbox
- Change loader

## Closes issue(s)
#232 

### When adding a completed task
Add-completed-task

### When removing a completed task
remove-comleted-task

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/pages/DailyGardening/index.tsx [hunk 1]
```diff
@@ -69,35 +65,30 @@ export const DailyGardening = () => {
...
 
   const rules = useMemo(
     () => gardenDataApi.response?.rules ?? [],
     [gardenDataApi]
   );
 
-  const completedTasks = useMemo(() => {
-    return currentCompletedTasks;
-  }, [currentCompletedTasks]);
-
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:88]: Remove this memo and use state directly

File: src/pages/DailyGardening/index.tsx [hunk 2]
```diff
@@ -140,8 +131,40 @@ export const DailyGardening = () => {
...
+      }
+    },
+    [completeTaskHandler, removeCompletedTask]
+  );
+
   useEffect(() => {
     if (completedTaskApi.status === "succeeded") {
+      // MEMO: Return also deleted task id and not just user account data
       const { balance: newCoinBalance } = completedTaskApi.response.user;
       setCurrentCompletedTasks((v) => [
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:169]: Currently api return s only user account updated data, would be nice to have the deleted task data too at least the id

File: src/pages/DailyGardening/index.tsx [hunk 3]
```diff
@@ -232,41 +255,38 @@ export const DailyGardening = () => {
...
-                          disabled={completedTaskApi.status === "loading"}
-                          // TODO: Implement UNDO
-                          // onDelete={() => handleDelete(rule)}
-                          // deleteIcon={<UndoIcon />}
-                        >
-                          {rule.name}
-                        </Button>
+                      <div key={rule._id}>
+                        <div className={styles.taskCheckboxWrapper}>
+                          <FormControlLabel
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:260]: Change button to checkbox

File: src/pages/DailyGardening/index.tsx [hunk 4]
```diff
@@ -232,41 +255,38 @@ export const DailyGardening = () => {
...
+                                checked={isRuleCompleted(rule._id)}
+                              />
+                            }
+                            label={rule.name}
+                          />
+                          {lastClicked === rule._id &&
+                            isTaskHandlerApiLoading && (
+                              <CircularProgress size={24} />
+                            )}
+                        </div>
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:281]: Showing loader to the side of each task

File: src/pages/DailyGardening/index.tsx (hunk 1 above)
Comment (kome12) [reply src/pages/DailyGardening/index.tsx:88]: Thank you!

File: src/pages/DailyGardening/index.tsx (hunk 2 above)
Comment (kome12) [reply src/pages/DailyGardening/index.tsx:169]: We can fix this on the backend after the demo.

**Review by kome12 — APPROVED** [review]
LGTM!

**Review by conalli — APPROVED** [review]
LGTM

**Pull request merged**